Cellular Notebooks: Parameter Plot, LFP Plots and Feature Extraction: Voltage and Current Clamp

darshanmandge opened the pull request:
Hi,
I created 7 Cellular Notebooks under these: 
- Parameter Plot, 
- LFP Plots and 
- Feature Extraction: Voltage and Current Clamp

I had to put the data locally for now, but I can replace them with links to the resources so that we only have notebooks here. The metadata is missing. I need to get the nexus metadata somehow and put it for each notebook

darshanmandge commented:
Hi @danielaegassan !
Yes, the comments were addressed in the last commit. 
I will now resolve conflicts and merge changes from main branch in the latest commit

darshanmandge commented:
@danielaegassan @MWolfR Should we merge this PR? I have merged all changes from the main branch to this PR branch, updated the analysis_info.json and added readme for all my notebooks.

@g-bar Will this branch work with the current implementation on staging?

danielaegassan commented:
@darshanmandge checking one more time for the buggy parts and will get back to you

danielaegassan commented:
@darshanmandge as far as I can see the notebooks that use nrnivmodl haven't been changed to use path objects instead of just strings.  So this is still buggy on my side.  I think this should be changed before merging.

g-bar commented:
> @g-bar Will this branch work with the current implementation on staging?

Seems to work fine.

MWolfR commented:
Let's keep in mind that these notebooks are eventually supposed to be running on the platform, but also be useful for local download. How is this one making this compromise?

darshanmandge commented:
> Let's keep in mind that these notebooks are eventually supposed to be running on the platform, but also be useful for local download. How is this one making this compromise?

For running the notebooks on platform, 
1. I will replace the model folder download option with a script to download the platform model resources. We have a way to access via Nexus but if we move to the new database, this will change.
2. For compiling of the mod files, depending on the location where the model gets downloaded on platform, I will update the path in the notebooks.

For local runs: I specify where the model files have to be kept within notebooks and README. So, user should be able to run them easily on a PC.

MWolfR commented:
@darshanmandge
One of the notebooks says:

> Download the e-model folder from [here](https://openbraininstitute.sharepoint.com/:f:/s/OBI-Scientificstaff/Ei3QIGh3JFFHkRkY7LRyTpEBm8eUll7HGyusulkPavf5SA?e=7ihV8H) in OneDrive `OBI -> Scientific staff -> Documents -> Data -> Analysis notebook data -> cadpyr_emodel`

That does not work for a public resource. That link is only accessible by us. Can you instead specify how to download e-models from the platform? Remember: The idea is that it analyzes the artefacts the way the "download" functionality serves them!

darshanmandge commented:
Yes, currently, only a JSON file is downloaded when downloading an e-model or me-model. I have created an [issue](https://github.com/openbraininstitute/prod-singlecell-simulation/issues/52#issue-2868261452) to download all the artefacts required for the notebooks when the user clicks the download button. 

As a workaround, I can create a download option from Blue Brain Open Data using AWS CLI if you want.

darshanmandge commented:
Hi @MWolfR,

> I think the only solution is to provide clear instructions how to download and assemble the required files from the platform. Both in the README and the main notebook.

I have added instructions to assemble and download data for the main `emodel` folder [README](https://github.com/openbraininstitute/obi_platform_analysis_notebooks/tree/cells_dm1/Cellular/emodels#downloading-data-for-notebooks), individual notebooks and their README, e.g. cadpyr_showcase [README](https://github.com/openbraininstitute/obi_platform_analysis_notebooks/blob/cells_dm1/Cellular/emodels/cadpyr_showcase/README.md#use) and [notebook](https://github.com/openbraininstitute/obi_platform_analysis_notebooks/blob/cells_dm1/Cellular/emodels/cadpyr_showcase/analysis_notebook.ipynb). I have removed the testing data download links which were only accessible to us. 
Thus, the user can get the data from the Platform or the Blue Brain Open Data. One of the notebooks requires accessing data from [channelpedia](https://github.com/openbraininstitute/obi_platform_analysis_notebooks/tree/cells_dm1/Cellular/efeature_extraction/voltage_clamp_ephys_extraction#use). For accessing AWS data, I had to add `aws-cli` to the requirements. I have commented on this [issue](https://github.com/openbraininstitute/prod-launch-notebook/issues/13#issuecomment-2684437963) to make it available to all users of JupyterHub.

@danielaegassan: You won't have any problems with compiling mod files and running the LFPy notebooks, as the notebooks now download the data directly into a folder without spaces.

@ilkilic @AurelienJaquier: Your notebooks were also using emodel data. I have replaced the private zip you used with data from the platform/AWS as discussed. For impedance and currentscape data, I replaced the ball-and-stick model with a detailed morphology model.

AurelienJaquier commented:
@darshanmandge  After discussion with Rajnish, I leanred that channelpedia data are actually accessible from aws also. So you can get the ion channel traces for the notebook with the following command:
`aws s3 cp --no-sign-request s3://openbluebrain/Experimental_Data/Electrophysiological_recordings/Ion_channels/Kv/Kv1.1/rCell9430.nwb ./`

darshanmandge commented:
Thanks @AurelienJaquier! I have made aws download the default option for the VClamp (ion channel) [notebook](https://github.com/openbraininstitute/obi_platform_analysis_notebooks/blob/cells_dm1/Cellular/efeature_extraction/voltage_clamp_ephys_extraction/analysis_notebook.ipynb). I have also updated the README and the JSON.

darshanmandge commented on Cellular/emodels/lfpy_simulations/active_emodel_synapses/analysis_info.json:
Is this fine "matplotlib>=3.10.0" to make it work for newer versions of matplotlib?

darshanmandge commented on Cellular/emodels/lfpy_simulations/active_emodel_Iclamp/analysis_info.json:
I have made it `matplotlib=>3.10.0` so that the notebook can work with newer versions of matplotlib too. I hope it is okay with you.

darshanmandge commented on Cellular/emodels/lfpy_simulations/active_emodel_Iclamp/analysis_notebook.ipynb:
As discussed in the chat this was an old error and was resolved. The floating text "edit path here" as discussed on chat was removed

darshanmandge commented on Cellular/emodels/lfpy_simulations/active_emodel_synapses/analysis_notebook.ipynb:
Thanks @danielaegassan. I have added the following text at the start of all the `emodel` folder notebooks:
```
# if you don't have the venv path try the last comment 
# try by replacing >> {venv_path}/bin/nrnivmodl << with >> nrnivmodl << in the commands below

venv_path=path/to/venv
emodel_path="./cadpyr_emodel"

# compile mechanisms before before calling LFPy
!'{venv_path}bin/nrnivmodl' {model_path}/mechanisms

#Alternatively 
import os
os.system(f"{venv_path}/bin/nrnivmodl {emodel_path}/mechanisms")

# You can also try by replacing >> {venv_path}/bin/nrnivmodl << with >> nrnivmodl << in both the commands above
```
I added the following comment at the begining so that if the usersdon't know the `venv_path` when running on platform, user can just try using:

```
# if you don't have the venv path try the last comment 
# try by replacing >> {venv_path}/bin/nrnivmodl << with >> nrnivmodl << in the commands below
```
